### ISSUE TITLE
Feature request: TIA Email Invoice Updates (TL)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#2633]
### As a TIA reviewer, I want the system to email the folder invoice from our group email account. I also want the system to make it easier to provide standard payment information to the applicant.

**Ready for Development**

**Stakeholders:** Nazlie, Sangeeta
**DTS Manager:** Tracy

**Acceptance Criteria:**
- On Initial Intake Process, 'Email Invoice' attempt is available
- When 'Email Invoice' attempt is saved, the system sends the folder applicant 1 email
    - Email body template is [here](https://github.com/cityofaustin/atd-data-tech/files/4854252/Email.Invoice.Template.v1.0.docx)
    - Each unpaid invoice is attached

**Comment by TracyLinder** [comment]
**Email Body:**

Hello,

The Austin Transportation Department (ATD) has implemented a City Council approved fee relating to the review of Transportation Impact Analyses (TIA). For further information, please refer to the fee structure located here: www.austintexas.gov/sites/default/files/files/Transportation/Transportation_Development_Services/FY20_Fee_Structure.pdf

Attached is an invoice pertaining to ATD’s TIA review of this development. Please note that these fees need to be paid before or at the first submittal of the TIA.

The invoice may be paid either: 

With the cashier’s office at One Texas Center, 505 Barton Springs Rd, 1st floor 
 Mon – Fri 8:00AM – 4:00PM

OR

Online at Austin Build + Connect

For questions or concerns, please reply to this email.

Thank you,

Transportation Development Services Division
Austin Transportation Department
901 S. MoPac Expressway, Building 5, Suite 300
Austin, TX  78746

**Comment by TracyLinder** [comment]
Prioritization From Customer:

![image.png](https://images.zenhubusercontent.com/5cca1e522b8dfc0ee14dedf5/5a300218-c6e0-4c91-9690-dbdc4de72c2e)

**Comment by Nadin-Nader** [comment]
@TracyLinder Spoke to Sangeeta via chat and asked if i could get a copy of the actual current email that is sent to customers so that i could research this further. Researched to find the email in AMANDA but was not able to find where it sleeps. I had George look for it as well and he could not find the email in AMANDA either.
Body of the email is below: I will ask Gita how best to research it.

Hello,

The Austin Transportation Department (ATD) has implemented a City Council approved fee relating to the review of Transportation Impact Analyses (TIA). For further information, please refer to the attached fee structure.

Also attached is an invoice pertaining to ATD’s TIA review of this development. Please note that these fees need to be paid before or at the first submittal of the TIA.

The invoice may be paid either: 

With the cashier’s office at One Texas Center, 505 Barton Springs Rd, 1st floor 
 Mon – Fri 8:00AM – 4:00PM

OR

Online at Austin Build + Connect

For questions or concerns, please reply to this email.

Thank you,

Transportation Development Services Division
Austin Transportation Department
901 S. MoPac Expressway, Building 5, Suite 300
Austin, TX  78746

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader 
[Email Invoice Template v1.0.docx](https://github.com/cityofaustin/atd-data-tech/files/4814599/Email.Invoice.Template.v1.0.docx)

We need these clarifications to continue on #2633

1.	Email Invoice feature exists in the current system for TIA Folder or not? Or we do it manually using the email option in fee/document tab?
2.	If it does not exist, do you want to it under “Approved” attempt result in Submittal intake Process i.e, when Approved and it updates the folder status to “Approved” as well as send this email to recipient with the message and from address will be “transportationreview@austintexas.gov” 
3.	W.r.t email template please find a sample format for your reference and provide us accordingly.

**Comment by TracyLinder** [comment]
- Option 1: Enable email on fee tab
- Option 2: 'Email Invoice' attempt on Initial Intake Process 

- [ ] @Sreelatha-BlazeKannan is checking to see if option #1 is possible
- [ ] Tracy needs to get fee structure attachment from Sangeeta


[Email Invoice Template v1.0.docx](https://github.com/cityofaustin/atd-data-tech/files/4854226/Email.Invoice.Template.v1.0.docx)

**Comment by TracyLinder** [comment]
**Email to Joan M, Nalizie:**

Hello,

I hope you are both doing well. We’re currently working on your request to automate the TIA invoice email feature (more info here). We will need the fee structure document that you’d like to be attached to the email that goes to the customer with the invoice. Are either of you able to send that to me? 

All the Best,

Tracy Linder, Product Manager

**Comment by Sreelatha-BlazeKannan** [comment]
Option 1 – Enable email in Fee tab would not work for TIA Folder invoice email updates. 

•	Reason 1 - From Email id is retrieved from the Valid User email id which can not be set to common email id
•	Reason 2 - Body message/subject cant be updated for TIA folder as this setup is for enterprise level

So, we would continue to work with Option 2 – Folder process new attempt “Email Invoice”.

**Comment by TracyLinder** [comment]
**Status:** Currently in development with LIT.

**Comment by TracyLinder** [comment]
**Status:** Began working this a few days ago. Updating to sprint 29.

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader  - We would require  the fee structure attachment from Sangeeta, Could you please get it for us. thank you!

**Comment by TracyLinder** [comment]
Acceptance criteria has been updated.

**Comment by TracyLinder** [comment]
[FY20 Fee Structure.pdf](https://github.com/cityofaustin/atd-data-tech/files/4991098/FY20.Fee.Structure.pdf)

@Sreelatha-BlazeKannan the fee structure is attached to this comment.

**Comment by Sreelatha-BlazeKannan** [comment]
@aishwarya2505  - Updated Template 
[Email Invoice Template v1.1.docx](https://github.com/cityofaustin/atd-data-tech/files/4991142/Email.Invoice.Template.v1.1.docx)

**Comment by TracyLinder** [comment]
**Status:** Work to continue into sprint 30.

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader - #2633 this is on hold as AIshwarya is working on #2631 changes.

**Comment by TracyLinder** [comment]
**Status:** Ash will begin working on this today.

**Comment by Nadin-Nader** [comment]
**Status:**
Aish is currently working on this

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader  - Aishwarya submitted DEV Commit  for configuration changes .

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder @aishwarya2505 Submitted SCTASK0093819 Request for CM DevOps to deploy the changes to DEV for commit 8f70db28256f

**Comment by Nadin-Nader** [comment]
@aishwarya2505 @Sreelatha-BlazeKannan I have created this lookup in DEV. Please verify and let me know.

**Comment by Sreelatha-BlazeKannan** [comment]
Technical Solution :  

**Alternative 1** :

When "Email Invoice" attempt chosen,  If we want to have all the invoice(s) of a particular folder to be e-mailed as one attachment in a single e-mail, we could create a new report from scratch. This new report path would ensure  all the invoice being printed page after page for a single folder = one e-mail to the applicant.

We can provide the link to the static text in our e-mail body or we can include in the report itself.

OR
**Alternative 2**:
When "Email Invoice" Attempt Chosen for each invoice number one email will be sent
This would include a batch job that inserts the report in the Folder as an attachment. In this case, I would be needing the attachment save path. We need this attachment to be stored in the folder temporarily. 
An "after procedure" can later be run post the batch job fires to send two attachments through an e-mail. 
The attachments from the folder can be deleted once the e-mail is sent. 

Please do let us know which solution do you prefer to go with. Thank you! 

If required, we all can have a meeting to discuss about this.

**Comment by Nadin-Nader** [comment]
@aishwarya2505 @Sreelatha-BlazeKannan i asked Nazlie and now im waiting on a response

**Comment by Nadin-Nader** [comment]
@aishwarya2505 @Sreelatha-BlazeKannan spoke to Sangeeta via chat and she would like to go with Alternative 1:

Alternative 1 :

When "Email Invoice" attempt chosen, If we want to have all the invoice(s) of a particular folder to be e-mailed as one attachment in a single e-mail, we could create a new report from scratch. This new report path would ensure all the invoice being printed page after page for a single folder = one e-mail to the applicant.

We can provide the link to the static text in our e-mail body or we can include in the report itself.

**Comment by aishwarya2505** [comment]
DEV Deployment Plan sent on 9/21/2020

The deployment plan has been sent prior to sending the demo on local host as we would like to test the flow in DEV first to make sure the e-mail and batch job work as expected. 

@TracyLinder @Nadin-Nader @Sreelatha-BlazeKannan

**Comment by Nadin-Nader** [comment]
@aishwarya2505 @Sreelatha-BlazeKannan I have sent this to the CM DevOps Team to deploy. The task # is SCTASK0097002.

**Status:** 
Waiting on CM DevOps Team to Deploy to DEV

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader - Waiting on DEV Move.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder @aishwarya2505 The task has been closed and its on DEV

**Comment by aishwarya2505** [comment]
Thank you @Nadin-Nader , I shall begin testing shortly.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Aishwarya will have to test and update the same.

**Comment by TracyLinder** [comment]
**Status:** Pending update from LaunchIT

**Comment by aishwarya2505** [comment]
@TracyLinder @Nadin-Nader @Sreelatha-BlazeKannan Tested this on DEV and have made further commits.

**Comment by aishwarya2505** [comment]
The changes are in DEV, I will be working with CTM to have this tested today since it involves the insertion of  a batch job. 

 @Nadin-Nader @TracyLinder @Sreelatha-BlazeKannan

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader @TracyLinder -
Aish was looking into this issue with Guna from CTM and have currently kept this on hold to make changes for #2983.

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader @TracyLinder - Sree, Subha & AIshwarya are working on this card.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder Meeting set for 11/11/2020 at 10am

**Comment by TracyLinder** [comment]
**Status:** We are not going to be able to deliver this before the code freeze on 11/20. This is going back into the backlog until V7 upgrade is complete.

**Comment by TracyLinder** [comment]
Prioritization Note: Not currently a priority. Moving to ice box.

**Comment by TracyLinder** [comment]
Still not a priority. Closing.
